pandas/core/nanops.py

```diff
@@ -200,12 +200,34 @@ def _get_fill_value(dtype, fill_value=None, fill_value_typ=None):
                 return tslibs.iNaT
 
 
-def _get_values(values, skipna, fill_value=None, fill_value_typ=None,
-                isfinite=False, copy=True, mask=None):
+def _maybe_get_mask(values, skipna, mask):
+    """ This function will return a mask iff it is necessary. Otherwise, return
+    None when a mask is not needed.
+    """
+    if (hasattr(values, 'dtype') and is_bool_dtype(values.dtype)):
+        # Boolean data cannot contain nulls, so signal via mask being None
+        return None
+
+    mask = _get_mask(values, mask, skipna=skipna)
```

Comment:
why can't you do this inside _get_mask itself? (the test)

Reply:
The `_get_counts()` function relies on `mask` providing basic shape information even when `skipna=False`. Unfortunately, creating an empty mask (eg, `mask = np.zeros(values.shape)`) to propagate this shape info for `nanmean`/etc negates the speed benefit to `nanall()`/`nanany()`, so two separate approaches are necessary.

Reply:
can we pass ``mask=None`` in those cases? I find this having to call 2 functions much more overhead than before.